h2/src/main/org/h2/mvstore/tx/TransactionMap.java

```diff
@@ -709,7 +732,7 @@ public DataType getKeyType() {
     private static class UncommittedIterator<K, X> extends TMIterator<K, X> {
 
         UncommittedIterator(TransactionMap<K, ?> transactionMap, K from, K to, boolean forEntries) {
-            super(transactionMap, from, to, transactionMap.getStatementSnapshot(), forEntries);
+            super(transactionMap, from, to, transactionMap.createSnapshot(), forEntries);
```

Comment:
It looks like `createSnapshot()` shouldn't be used here by the plain `UncommittedIterator` (only by `ValidationIterator`), because it may use other index in inconsistent state in the same query.

Reply:
On the other hand, maybe you right about READ UNCOMMITTED, since we are not gaining much in terms of performance - just skipping dependencies collection step?

Reply:
This level is not expected to be faster, it isn't about the performance. It is used to monitor activity of long-running transactions and in similar use cases. Actually in can be slower, because `COUNT(*)` has optimized version only for a `READ COMMITTED`.

I don't think that without a consistent snapshot the query will be faster, most likely the performance will be nearly the same.

Reply:
>This level is not expected to be faster, it isn't about the performance

Not sure who’s expectations you are referring to. 
In any case, to expect any level of consistency from that level is a sure way to insanity. For the purpose of monitoring you may want latest data possible, rather than semi-consistent snapshot, or the other way around, if snapshot is recent enough. You can argue both ways. You can only get truly consistent snapshot only for a standalone transaction run.

Reply:
I explained in other comment https://github.com/h2database/h2database/pull/2173#discussion_r334962008 that every single row can't be partially visible (visible for one expression and invisible for other one), or have different values during evaluation of different expressions in select list. Such possible situation is insane even for `READ UNCOMMITTED`.

Reply:
OK, l agree that consistency isn't ensured by this level anyway.